### ISSUE TITLE
Undo / redo highlight and selection of text

### ISSUE THREAD

**Issue by jsr-p** [ahrm/sioyek#633]
Hi 

Thanks for the great pdf viewer!

I was wandering whether it is possible to undo  & redo highlight + selection of text.
I tried to make a python extension doing this by manipulating the db file.
Unfortunately, I get an `SQL Error: database is locked` when trying to delete the row containing the latest hightlight, while the `sioyek` process is locking the sqlite3 database. 
Is there another way to implement this; maybe it is already implemented and I just can't see it in the documentation?

Thanks again.
- jsr-p

**Comment by ahrm** [comment]
You mean deleting a highlight? You can do that by clicking on a highlight and executing `delete_highlight` command or just pressing the `delete` key when the highlight is selected in the highlights list.

**Comment by jsr-p** [comment]
@ahrm yes, but without touching the mouse; for instance deleting the latest highlight by pressing `u`, and then being able to redo this highlight by pressing, say, `<C-r>`.

**Comment by ahrm** [comment]
You can open a list of highlights by pressing `gh` (the closest highlight should be automatically highlighted) and press `delete` after selecting the highlight you want to delete.

**Comment by jsr-p** [comment]
Ah, nice! 
But is it then possible to redo the highlight using some keybinding? For instance if the deletion wasn't intended.

**Comment by ahrm** [comment]
No, I don't think it is a common enough occurrence to implement. Worst case scenario we could just highlight it again, it's not like we are losing a lot of valuable data.

**Comment by jsr-p** [comment]
All right :smiley:
Thanks for the answers.

**Comment by NightMachinery** [comment]
@ahrm This `Delete` key doesn't seem to work on macOS. Highlights show no sign of being selected either.

Adding a `remove_last_highlight` command and binding it to `u` is a good idea. Having a redo helps when accidentally pressing this command, but it's not that important.

**Comment by NightMachinery** [comment]
> You can open a list of highlights by pressing `gh` (the closest highlight should be automatically highlighted) and press `delete` after selecting the highlight you want to delete.

Does this actually work? I have tested it on the commit 779b52e36de9d596dc708ada045ea48af0c21c9d and it doesn't work until I click on the highlight using the mouse.
